crates/tabby/src/routes/completions.rs

```diff
@@ -26,11 +26,14 @@ use crate::services::completion::{CompletionRequest, CompletionResponse, Complet
 pub async fn completions(
     State(state): State<Arc<CompletionService>>,
     TypedHeader(MaybeUser(user)): TypedHeader<MaybeUser>,
+    TypedHeader(user_agent): TypedHeader<headers::UserAgent>,
     Json(mut request): Json<CompletionRequest>,
 ) -> Result<Json<CompletionResponse>, StatusCode> {
     if let Some(user) = user {
         request.user.replace(user);
     }
+    request.user_agent = Some(user_agent.to_string());
+
     match state.generate(&request).await {
```

Comment:
```suggestion
    match state.generate(&request, &user_agent.to_string()).await {
```

Instead of adding `user_agent` into request, I recommend modify the CompletionService's api interface - adding it as a parameter of `generate`.

Reply:
Sure, I'll fix rn